pkg/tfbridge/provider.go

```diff
@@ -1139,6 +1139,11 @@ func (p *Provider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*pulum
 		changes = pulumirpc.DiffResponse_DIFF_SOME
 	}
 
+	properties := []string{}
+	for key := range uniqueProps {
+		properties = append(properties, key)
+	}
```

Comment:
You can then keep the more descriptive name `properties` and just pass `Properties: toSlice(properties),` directly in the return call.

```suggestion
	toSlice := func(m map[string]struct{}) []string {
		arr := make([]string, 0, len(m))
		for k := range m {
			arr = append(arr, k)
		}
		sort.Strings(arr)
		return arr
	}
```